### PR TITLE
fix: Remove dangling `snapErrors` state property

### LINKED ISSUE #26281: Error: No metadata found for 'snapErrors'
Sentry Issue: [METAMASK-X62M](https://metamask.sentry.io/issues/4344650755/?referrer=github_integration)

```
Error: No metadata found for 'snapErrors'
  at Array.reduce (<anonymous>)
  at handler (app/scripts/lib/ComposableObservableStore.js:70:30)
  at async Promise.all (index 0)
...
(6 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by FrederikBolding**
## **Description**

Adds a line to the `120.2` migration to also remove a dangling state property in the `SnapController`, that is causing sentry errors.

This property is likely dangling due to https://github.com/MetaMask/metamask-extension/pull/26280

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26282?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/26281

### CODE REVIEW

File: app/scripts/migrations/120.2.ts
Comment (Gudahtt) [inline app/scripts/migrations/120.2.ts:30]: Use of `any` is disallowed by our TypeScript guidelines. The `unknown` type helps ensure we validate persisted state before using it, preventing further errors from appearing in migrations due to unexpected state corruption. Lets keep the `unknown` type here:

```suggestion
function transformState(state: Record<string, unknown>) {
```

See here on why we should avoid `any`, and on alternatives: https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#avoid-any

Comment (Gudahtt) [reply app/scripts/migrations/120.2.ts:30]: I see that in this case, it's hiding a TypeScript error caused by is not knowing whether `state.SnapControlelr` is an object.

We could handle that case by reporting a Sentry error and returning state unmodified. Probably best not to clear it as we're doing with SelectedNetworkController, just in case it has some recoverable data despite being the wrong type (e.g. what if we stringified it accidentally, then we could error and fix with a migration a later hot fix), but it's your call if you'd prefer console.warn + state clear for that case instead.

Comment (Gudahtt) [reply app/scripts/migrations/120.2.ts:30]: I see that there is the potential for unexpected corruption in one state to impact the other. This is not ideal, but maybe not a deal breaker because we don't expect ti to happen. It just complicates the cleanup effort if it does happen. We could avoid this complication by refactoring the migration to use two separate functions to mutate the state here, one per controller, so they act more independently. Or we could move them to separate migrations. Not a blocker but that would be an improvement I think.

Comment (FrederikBolding) [reply app/scripts/migrations/120.2.ts:30]: How would you prefer the typing of `delete state.SnapController.snapErrors` then?

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Requesting changes due to the missing validation and use of `any`. Looks great aside from that though! The other suggestions I made are non-blocking.

File: app/scripts/migrations/120.2.ts
Comment (Gudahtt) [reply app/scripts/migrations/120.2.ts]: Resolving because the `any` type has been removed.

We're still not reporting or fixing corrupted `SnapController` state. Typically we would in migrations, but I'm not sure how much that matters, I am not inclined to block on that.

**Review by Gudahtt — APPROVED** [review]
LGTM!

We might make a few more changes to this migration in future PRs (we have more corrupted state to fix), so we can consider some of the other suggestions I made in those later PRs.

**Review by adonesky1 — APPROVED** [review]
LGTM

**Pull request merged**
